benchmarking/blackbox_repository/simulated_tabular_backend.py

```diff
@@ -156,9 +156,46 @@ def _run_job_and_collect_results(
                 result[self.elapsed_time_attr] -= elapsed_time_offset
         else:
             results = all_results
+
+        # Makes sure that time is monotonically increasing which may not be the case due to numerical errors or due to
+        # the use of a surrogate
+        for i in range(1, len(results)):
+            results[i][self.elapsed_time_attr] = max(
+                results[i][self.elapsed_time_attr],
+                results[i - 1][self.elapsed_time_attr] + 0.001
+            )
+
         return status, results
 
 
+def make_surrogate(surrogate: Optional[str] = None, surrogate_kwargs: Optional[Dict] = None):
```

Comment:
Ah OK, so this would get us around these issues I had?

Reply:
Yes.